cmake/therock_external_source.cmake

```diff
@@ -0,0 +1,24 @@
+# Enables to build a subcomponent from a user provided source location.
+function(therock_enable_external_source package_name default_source_dir default_option)
+  string(TOUPPER ${package_name} _PACKAGE_UPPER)
+  string(REPLACE "-" "_" _PACKAGE ${_PACKAGE_UPPER})
+  option(THEROCK_USE_EXTERNAL_${_PACKAGE} "Use external ${package_name} source location" ${default_option})
```

Comment:
Instead of creating options here, this could be required explicitly by dropping https://github.com/ROCm/TheRock/pull/1237/commits/5124e71112cde1c5ab6a37fd525cdf71eefcb685.